index.js

```diff
@@ -22,26 +22,50 @@ class DetailsMenuElement extends HTMLElement {
     const summary = details.querySelector('summary')
     if (summary) summary.setAttribute('aria-haspopup', 'menu')
 
-    onOpen(details)
-
     details.addEventListener('click', shouldCommit)
     details.addEventListener('change', shouldCommit)
     details.addEventListener('keydown', keydown)
-    details.addEventListener(
-      'toggle',
-      () => {
-        if (!this.src) return
-        const loader: any = this.querySelector('include-fragment')
-        if (loader) {
-          loader.addEventListener('loadend', focusInput.bind(null, details))
-          loader.src = this.src
-        }
-      },
-      {once: true}
-    )
-
+    details.addEventListener('toggle', loadFragment, {once: true})
     details.addEventListener('toggle', closeCurrentMenu)
-    details.addEventListener('toggle', focusInput.bind(null, details))
+
+    const subscriptions = [onOpen(details)]
```

Comment:
Can we give `onOpen` a more specific name? Now that they are by each other `opOpen` looks ambiguous since `loadFragment` and `closeCurrentMenu` happen on open too. 

Also curious why is `onOpen` bound via subscriptions but the other events aren't? Should we use one or the other? Or is this change coming, hence the `[]`?

Reply:
I considered building a subscription for the other listeners but it felt a little artificial.

```js
const subscriptions = [
  focusOnOpen(details),
  {
    unsubscribe: () => {
      details.removeEventListener('click', shouldCommit)
      details.removeEventListener('change', shouldCommit)
      details.removeEventListener('keydown', keydown)
      details.removeEventListener('toggle', loadFragment, {once: true})
      details.removeEventListener('toggle', closeCurrentMenu)
    }
  }
]
```

The `focusOnOpen` behavior benefits from the subscription object because it's a self-contained behavior requiring several listeners working together with some shared state (e.g. `isMenu`).

Reply:
👍 Thanks for the explanation. The alternatives are indeed not really much better.